src/payload/addr.rs

```diff
@@ -0,0 +1,711 @@
+//! IP address resources.
+
+use std::{error, fmt};
+use std::net::{AddrParseError, IpAddr, Ipv4Addr, Ipv6Addr};
+use std::num::ParseIntError;
+use std::str::FromStr;
+#[cfg(feature = "serde")] use serde::{
+    Deserialize, Deserializer, Serialize, Serializer
+};
+
+
+//------------ Bits ----------------------------------------------------------
+
+/// The value of an IP address.
+///
+/// This private type holds the content of an IP address. It can be both an
+/// IPv4 and IPv6 address. It keeps the address internally as a 128 bit
```

Comment:
Because there is no way of distinguishing which one you have from just the bits, it really is both a v4 and a v6 address at the same type. I have rephrased the text, though.